### PR TITLE
RWA-686 pact test added for search task with work-type

### PR DESCRIPTION
**Pull request by serdaroguzhanakin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-686

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by alexbaeza — APPROVED** [review]
LGTM 👍

**Review by BenNzewi — COMMENTED** [review]
What about endpoints

1. SearchForCompletable
2. getTask

Will this also return workTypes? We will need to update these also in that case.

**Review by BenNzewi — COMMENTED** [review]
this fails local verification

Verifying a pact between wa_task_management_api and wa_task_management_api_search
  [Using File pacts/wa_task_management_api-wa_task_management_api_search.json]
  Given appropriate tasks are returned by criteria with work-type with warnings only
  Provider receives a POST /task request from a WA API
    returns a response which
      has status code 200 (FAILED)
      includes headers
        "Content-Type" with value "application/json; charset=UTF-8" (FAILED)
      has a matching body (FAILED)

**Review by BenNzewi — COMMENTED** [review]
this fails local verification

Verifying a pact between wa_task_management_api and wa_task_management_api_search
  [Using File pacts/wa_task_management_api-wa_task_management_api_search.json]
  Given appropriate tasks are returned by criteria with work-type
  Provider receives a POST /task request from a WA API
    returns a response which
      has status code 200 (FAILED)
      includes headers
        "Content-Type" with value "application/json; charset=UTF-8" (FAILED)
      has a matching body (FAILED)

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/model/TaskWithWorkType.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.provider.model;
+
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.task.WarningValues;
+
+import java.time.ZonedDateTime;
+
+public class TaskWithWorkType extends uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.task.Task {
```
[the remark this answers is not in the document]
Comment (serdaroguzhanakin) [reply src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/model/TaskWithWorkType.java:7]: I will solve it when I completed RWA-683

**Comment by serdaroguzhanakin** [comment]
> What about endpoints
> 
> 1. SearchForCompletable
> 2. getTask
> 
> Will this also return workTypes? We will need to update these also in that case.

will be done as part of RWA-687

**Comment by serdaroguzhanakin** [comment]
> this fails local verification
> 
> Verifying a pact between wa_task_management_api and wa_task_management_api_search [Using File pacts/wa_task_management_api-wa_task_management_api_search.json] Given appropriate tasks are returned by criteria with work-type with warnings only Provider receives a POST /task request from a WA API returns a response which has status code 200 (FAILED) includes headers "Content-Type" with value "application/json; charset=UTF-8" (FAILED) has a matching body (FAILED)

I'm working on this issue

**Pull request closed without merging**